src/zmqsubscriber/zmqsubscriber.go

```diff
@@ -46,15 +47,14 @@ func NewZMQSubscriber(host string, port string) (*ZMQSubscriber, error) {
 		}
 	}
 
-	connectionString := "tcp://" + host + ":" + port
-	if err = socket.Connect(connectionString); err != nil {
-		return nil, fmt.Errorf("could not connect ZMQ subscriber to '%s': %s", connectionString, err)
+	if err = socket.Connect(zmqAddress); err != nil {
+		return nil, errors.Errorf("could not connect ZMQ subscriber to '%s': %s", zmqAddress, err)
 	}
 
-	log.Printf("ZMQ subscriber successfully connected to %s", connectionString)
+	log.Printf("ZMQ subscriber successfully connected to %s", zmqAddress)
 
-	incomingTx := make(chan types.Transaction)
-	incomingBlocks := make(chan types.Block)
+	incomingTx := make(chan types.Transaction, 1<<8)
+	incomingBlocks := make(chan types.Block, 1<<8)
```

Comment:
What is the rational for this change? Is there a reason why it's exactly 256? Does it make sense for the Blocks and Tx channels to be the same size?

Reply:
I must admit I didn't take proper notes on this change myself..

This loop will stall on channel write if either of the channels has no reader: https://github.com/0xB10C/bademeister-go/blob/67f0e2778587c2f22138162103acbd1116e72fff/src/zmqsubscriber/zmqsubscriber.go#L123 

With `GetRawMempoolVerbose()` it can stall for a long time. 

The main problem with that is shifting timestamps.

There is no real reason why it is 256. The number should be higher than the maximum number of txs that can arrive while the read loop is stalled. 

For blocks, 256 is probably way higher than necessary in mainnet, In regnet it could make sense. 

We should actually use `select` when writing to the channel and abort with error if the write is blocked. Otherwise we'll always have a risk of incorrect timestamps.